### ISSUE TITLE
valuation error

### ISSUE THREAD

**Issue by wangmxzyxwxg** [georgeNakayama/AIpparel-Code#22]
When I use the validation set for verification, I encounter such a problem: the edge prediction of the panel is incorrect, and self-intersection occurs, resulting in the final simulation failure. What could be the reason for this? Why do neural network predictions have such problems?

<img width="854" height="353" alt="Image" src="https://github.com/user-attachments/assets/4f57623f-620f-476e-aa2f-247496d064d9" />

**Comment by phphuc612** [comment]
TL;DR: The current training pipelines of these methods do not impose any constraints related to the sewing simulation process. A sewing pattern generated by these models does not guarantee a successful 3D reconstruction.

This is a known limitation in most works that approach pattern-drawing, such as DressCode, SewFormer, SewingLDM, and even AIpparel. Although these models can generate visually plausible sewing patterns, they are not trained with any physics- or simulation-aware constraints, so the predicted patterns may still fail when converted into 3D meshes. I recommend checking the simulation success rate or the sampling rate required to obtain a valid result for each method.

The pipeline from 2D pattern to 3D garment using GarmentCode can be broken into two major steps:

- Boxmesh construction: each predicted sewing pattern must be checked for valid stitching, degenerated triangles or invalid topology. These checks ensure the panels can be converted into a boxmesh, which places all panels around the SMPL body with stitching threads.  
--> If this step fails, you cannot obtain any mesh.
- Sewing simulation: the boxmesh is then passed to a simulation engine (e.g., NVIDIA Warp) to (1) collapse stitching threads so that panels sew together, and (2) simultaneously drape the resulting garment onto the SMPL as a collider.  
--> This step may fail with "intermediate" output mesh. The causes can be:  (1) self-collision within the garment, (2) inter-collision with the SMPL, and (3) simulation timeout, mostly due to unstable equilibrium where the garment cannot properly settle on the SMPL. These are resulted from the improper placing panels in the first step (creating boxmesh), and sometimes due to the simulation rules of GarmentCode. 
  
A recent work, ChatGarment, avoids direct pattern drawing. Instead, it relies on GarmentCode to ensure valid pattern by sampling patterns that pass sanity checks. However, in exchange for robustness, the results tend to be less diverse and offer less fine-grained control compared to pattern-drawing models.

**Comment by wangmxzyxwxg** [comment]
That is indeed the case. Theoretically, approaches that directly generate patterns offer better diversity; however, the lack of enforced simulation constraints leads to stitching failures in many instances. Yet, the simulation process itself is complex, and incorporating it into network training is a significant challenge.

Actually, one thing I am curious about is the approach of reducing 2D patterns into 1D sequences of edges and directly generating this 1D information. In doing so, wouldn't the spatial relationships inherent to the 2D structure be lost? Moreover, given the complex stitching relationships that exist among the panels, modeling this global spatial context is far more difficult than simple 2D spatial modeling.

**Comment by phphuc612** [comment]
Thanks for the discussion. From my research, efforts to turn the sewing pattern into a compact representation, which are aimed for diffusion-based and vae architecture, can be found in recent works: SewingLDM, GarmentDiffusion, GarmentImage, and GarmentX. However, I have not fully validated their actual capability to turn sewing pattern into 3D mesh. You may refer to these works as reference points or baselines for your alternative representations.
